pkg/chaosctl/common/common.go

```diff
@@ -408,17 +408,46 @@ Please check network policy / firewall, or see FAQ on website`, daemon.Name, dae
 }
 
 // NewChaosDaemonClientLocally would create ChaosDaemonClient in localhost
-func NewChaosDaemonClientLocally(port int, caCert string, cert string, key string) (daemonClient.ChaosDaemonClientInterface, error) {
+func NewChaosDaemonClientLocally(port int) (daemonClient.ChaosDaemonClientInterface, error) {
 	if cli := mock.On("MockChaosDaemonClient"); cli != nil {
 		return cli.(daemonClient.ChaosDaemonClientInterface), nil
 	}
 	if err := mock.On("NewChaosDaemonClientError"); err != nil {
 		return nil, err.(error)
 	}
-
-	cc, err := grpcUtils.CreateGrpcConnection("localhost", port, caCert, cert, key)
+	config, err := getTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+	cc, err := grpcUtils.CreateGrpcConnectionFromRaw("localhost", port, config.caCert, config.cert, config.key)
 	if err != nil {
 		return nil, err
 	}
 	return daemonClient.New(cc), nil
 }
+
+func getTLSConfig() (*rawTLSConfig, error) {
+	var cfg rawTLSConfig
+	restconfig, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(restconfig)
+	if err != nil {
+		return nil, err
+	}
+	secret, err := kubeClient.CoreV1().Secrets("chaos-testing").Get("chaos-mesh-daemon-client-certs", metav1.GetOptions{})
```

Comment:
It seems this place hardcoded the namespace 😕 Also maybe we could avoid magic words

Reply:
Absolutely! I was wondering if there's a way to automatically get chaosmesh namespace rather than setting it to a fixed value. Have got any ideas on how to do that?

Reply:
Maybe you could try to get the pod `chaos-daemon`, then to get its namespace. A reference could be: https://github.com/chaos-mesh/chaos-mesh/blob/master/pkg/chaosctl/cmd/logs.go#L90. Hope that be of help 😀

Reply:
https://github.com/chaos-mesh/chaos-mesh/blob/6ee0cb9f67c671eefb9110a99e021a260e2f5ac7/pkg/chaosctl/cmd/logs.go#L99

If I haven't got that wrong, it seems we still need to specify ns. I can see that code generation will definitely be a viable method but I'm afraid that'll be an overkill.

Reply:
I'm not so sure about this, but it seems normally when `clusterscoped` is set to `true`, arguments about namespace would be ignored. Although it seems only STRRL is using `chaosctl logs`, since he hasn't changed this part, maybe we could get chaos components as we need this way.
https://github.com/chaos-mesh/chaos-mesh/blob/f8c3fccd8e7317667b813973bda27c4748a0caa7/pkg/config/controller.go#L61-L63

Besides, I found this parameter (but I haven't tested it yet 😿 ) which seems is what we need.
https://github.com/chaos-mesh/chaos-mesh/blob/f8c3fccd8e7317667b813973bda27c4748a0caa7/pkg/config/controller.go#L72-L73

Reply:
That does not seem to work since they are both empty string 🤦

Reply:
ok then just add a todo 🤣